railties/lib/rails/commands/rake_proxy.rb

```diff
@@ -0,0 +1,38 @@
+require 'rake'
+
+module Rails
+  module RakeProxy
+
+    RAKE_TASKS_HELP_MESSAGE = <<-EOT
+In addition to those, you can now run the rake tasks as rails commands:
+
+    EOT
+
+    def write_rake_tasks_help_message
+      puts RAKE_TASKS_HELP_MESSAGE
+    end
+
+    def write_rake_tasks
+      width = rake_tasks.map { |t| t.name_with_args.length }.max || 10
+      rake_tasks.each do |t|
+        printf("#{Rake.application.name} %-#{width}s  # %s\n", t.name_with_args, t.comment)
+      end
+    end
+
+    def rake_tasks
+      require_application_and_environment!
+      Rake::TaskManager.record_task_metadata = true
+      Rake.application.instance_variable_set(:@name, 'rails')
```

Comment:
Unfortunately there is no other way to set the name, as rake doesn't expose the setter.
I can submit a pull request to rake project maintainer, if you think it makes sense to expose the setter.